bloom-housing/bloom
# fix: add better seeded data

# Pull Request Template

## Issue Overview

This PR addresses #3466 


## Description

As part of the typeORM -> Prisma changeover we need to re-think how data is seeded for both manual and automated testing. This PR creates a new pattern compared to the current backend by doing the following:

- Setup the seed file to do different things based on environment:
  - Prod: Only set the minimum fields needed to have a running environment such as a jurisdiction
  - Stage: Set up data that would make a demo (staging) site usable by a user. It sets up all of the mandatory fields along with mock tables that are consistent every time it is run. Lots of this data is setup from objects with a name and a diverse set  of values.
  - Dev: Sets up values to be used in local testing with completely random values. Names are random and a mix of values are randomly chosen.
- For all environments a jurisdiction name can be passed in the command by adding `--jurisdictionName <name>` when running the command.
  - dev environment is setup with the jurisdiction name of "Bloomington" so that running it locally you can have the name set in the .env file
- For each table type there is a factory to generate rows into that table. Everything that can be randomized is along with the ability to override values.
- The factories are used for all the seed setups (with different override values sent in all cases) and within the tests. The randomized values are then used to validate the fields rather than relying on a set value

## How Can This Be Tested/Reviewed?

After `yarn install` in the `backend_new` directory you can run the following commands and look to see what is in the DB.
`yarn db:setup` - setups the database with completely random values
`yarn db:setup:staging` - setups the database with the pre-defined staging values



## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
